cmd/flashback/main.go

```diff
@@ -299,10 +299,24 @@ func main() {
 		workerStates := make([]nodeWorkerState, len(nodes))
 
 		for i, n := range nodes {
-			session, err := mgo.Dial(n.url)
+			dialInfo, err := mgo.ParseURL(n.url)
+			panicOnError(err)
+			session, err := mgo.DialWithInfo(dialInfo)
 			panicOnError(err)
 			session.SetSocketTimeout(time.Duration(socketTimeout))
 			defer session.Close()
+			if dialInfo.Username != "" && dialInfo.Password != "" {
```

Comment:
Thanks @tredman, you're right - this is not required and is handled already as you mentioned. I've removed this block of code and successfully tested.

Also, I noticed that if auth is enabled but you don't specify any, flashback will run but with 100% errors. I've added an error handler for this case so we fail fast instead of running a full replay for no reason. Here is example output of this case:

```INFO 2016/11/26 17:32:17 logger.go:50: Started replaying ops by time with speedup of 1.000000
INFO 2016/11/26 17:32:17 logger.go:55: Worker #5 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #0 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #6 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #2 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #3 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #4 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #1 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #8 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #7 report for duty
INFO 2016/11/26 17:32:17 logger.go:55: Worker #9 report for duty
\ERROR 2016/11/26 17:32:17 logger.go:60: [default] not authorized to execute op - type:query,database:testdb1
ERROR 2016/11/26 17:32:17 logger.go:60: [default] not authorized to execute op - type:query,database:testdb2
ERROR 2016/11/26 17:32:17 logger.go:60: [default] not authorized to execute op - type:query,database:testdb1
ERROR 2016/11/26 17:32:17 logger.go:60: [default] not authorized to execute op - type:update,database:testdb1
ERROR 2016/11/26 17:32:17 logger.go:60: [default] not authorized to execute op - type:insert,database:testdb2
ERROR 2016/11/26 17:32:17 logger.go:60: [default] not authorized to execute op - type:update,database:test1
```